src/js/core/widget/core/MasterOnOffSwitch.js

```diff
@@ -0,0 +1,265 @@
+                element.classList.add(classes.WIDGET);
+
+                self._ui.onOff = onOff;
+                self._ui.labelOnOff = label;
+                self._ui.labelTextOnOff = labelText;
+        
+                // create instance of widget
+                ns.widget.OnOffSwitch(onOff);
+
+				return element;
+			};
+
+			/**
+			 * Initiate widget
+			 * @method _init
+			 * @param {HTMLElement} element
+			 * @protected
+			 * @member ns.widget.core.MasterOnOffSwitch
+			 * @instance
+			 */
+			prototype._init = function (element) {
+				var self = this;
+
+				self._ui.input = element.querySelector("input.ui-on-off-switch");
+				// set initial look
+				onChangeMasterOnOff(self);
+			};
+
+			function onChangeMasterOnOff(self) {
+				var onOff = self._ui.onOff,
+					label = self._ui.labelOnOff,
+					labelText = self._ui.labelTextOnOff;
+
+				if (onOff.checked) {
+					label.classList.add("ui-on-off-label-active");
+					labelText.innerHTML = "On";
+				} else {
+					label.classList.remove("ui-on-off-label-active");
+					labelText.innerHTML = "Off";
+				}
+				self._disableAllOnOff(!onOff.checked);
+			}
+
+			/**
+			 * Binds events to widget
+			 * @method _bindEvents
+			 * @protected
+			 * @member ns.widget.core.MasterOnOffSwitch
+			 * @instance
+			 */
+			prototype._bindEvents = function () {
+				var self = this,
+					onOff = self._ui.onOff,
+					_onChangeMasterOnOff = onChangeMasterOnOff.bind(null, self);
+
+				onOff.addEventListener("change", _onChangeMasterOnOff);
+
+				self._onChangeMasterOnOff = _onChangeMasterOnOff;
+			};
+
+			prototype._unbindEvents = function () {
+				var self = this,
+					onOff = self._ui.onOff;
+
+				onOff.removeEventListener("change", self._onChangeMasterOnOff);
+
+				self._onChangeMasterOnOff = null;
+			};
+
+			/**
+			 * Disables / enables on/off switches on list
+			 * @method _disableAllOnOff
+			 * @param {boolean} disabled
+			 * @member ns.widget.mobile.Listview
+			 * @protected
+			 */
+			prototype._disableAllOnOff = function (disabled) {
+                var targetElement = document.getElementById(this.options.target),
+                    allOnOff = [],
+					ui = this._ui,
+					onOffSwitch = null;
+
+                if (targetElement) {
+                    allOnOff = [].slice.call(targetElement.querySelectorAll(".ui-on-off-switch")),
+                    allOnOff.filter(function (item) {
+                        return item !== ui.onOff;
+                    }).forEach(function (onOff) {
+                        onOffSwitch = ns.widget.OnOffSwitch(onOff);
+                        if (disabled) {
+                            onOffSwitch.disable();
+                        } else {
+                            onOffSwitch.enable();
+                        }
+                    });
+                } else {
+                    ns.warn("MasterOnOffSwitch: indicated target element (" + this.options.target + ") not found");
+                }
```

Comment:
Indent looks weird.
Is the code linting working correctly on travis?

Reply:
Miałem niepoprawną konfigurację eslint in IDE.
Done.